examples/nextjs-csr/src/pages/logged-in.tsx

```diff
@@ -0,0 +1,81 @@
+import Link from "next/link";
+import { useEffect, useState } from "react";
+
+type UserInfoRes = {
+  sgid?: string;
+  userInfo?: Record<string, string>;
+  state?: string;
+};
+
+const LoggedIn = () => {
+  const [data, setData] = useState<UserInfoRes | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState("");
+
+  useEffect(() => {
+    const getUserInfo = async () => {
```

Comment:
Currently, the `userinfo` step and callback handling steps are performed separately, which seems to be different from what we agreed upon in the backend for frontend flow (https://app.gitbook.com/s/pBW92htBuXTrDYoKovQ2/~/changes/9/learn-the-basics/integration-patterns/backend-for-single-page-app-spa-frontend-bff). Could you elaborate on the rationale for splitting them?

Reply:
hmm i was under the impression that we were planning to perform them separately because of the [lucid chart diagram](https://lucid.app/lucidchart/477b82f6-19d2-4010-a8af-06e12688bd3b/edit?invitationId=inv_21f0e528-8e2a-40b7-8546-6ebbc3f1f617&page=0_0#) for the demo app that we did + Anty's Express demo app also separates them. 

I'll check in with Wira as he drew up the diagram

Reply:
Discussed with Wira - separating the userinfo and callback steps are more suited for session-based authentication but the diagram in the GitBook bff page is more generic and does not describe a userinfo endpoint. this is so that the RPs can decide whether they would like to employ session-based or token-based authentication for their app. 

do you think it's important for the app to be in-sync with the diagram for client authentication? (currently all our CSR examples split them)

Reply:
gotcha, thank you for clarifying! 

i don't think it's a big deal because both flows are secure, and I think it's fine to go with the flow with the split routes because that is what Anty and Wira have already agreed on. 

re: the inconsistency between the diagram and the code example, the diagram does refer to a userinfo endpoint. In an ideal world the example flow would be consistent with the diagram but I think the differences are small, and we can always fix the diagram later so let's just roll with it for now. What do you think?

Reply:
The differences in the diagram are on purpose because I didn't want to suggest a client authentication mechanism within the flow (in our example, we are using session-in-cookies based authentication and hence, the routes are different. I told @raynerljm to proceed with the example because we can always update the diagram whenever necessary.

Reply:
yup - i completely agree that we can always fix the diagram in the future. let's proceed with this first :D thanks for the discussion!